### ISSUE TITLE
Access to local file system inside Docker container?

### ISSUE THREAD

**Issue by mikegehard** [SWE-agent/SWE-ReX#148]
I quickly looked through the docs but could find the answer so thought I'd ask here. If there is a better place, let me know.

I'm working to implement a human in the loop AI assisted TDD flow using Claude desktop and MCP servers: https://github.com/mikegehard/ai-assisted-agile-agents/issues/14

I'm at the point where I need to allow Claude to run commands locally and I wanted to use Docker containers for isolation from my local machine: https://github.com/mikegehard/ai-assisted-agile-agents/issues/13#issuecomment-2527823388

I was thinking about writing an MCP server but couldn't figure out how to attach the local file system into the container: https://github.com/mikegehard/ai-assisted-agile-agents/issues/14#issuecomment-2531470528

Do y'all have any docs that would show how that works? I'm guessing I could just include a Dockerfile in my project to build the container with a volume mount but thought I'd ask.

Thanks!

**Comment by klieret** [comment]
Hi @mikegehard ! Not sure if I understand your question completely, but it sounds like you're trying to mount local folders into the docker container? (like with [bind mounts](https://docs.docker.com/engine/storage/bind-mounts/))?

With swe-rex Docker deployment (see the [config docs](https://swe-rex.com/latest/api/deployments/config/#swerex.deployment.config.DockerDeploymentConfig)), you should be able to just pass your arguments. 

So I think something like this:

```
docker_config = DockerDeploymentConfig(
	image="...",
	docker_args=["--mount type=bind,src=<host-path>,dst=<container-path>"]
)
```

or similarly with volume and then

```
deployment = DockerDeployment.from_config(docker_config)
```

Does that answer your question?

**Comment by mikegehard** [comment]
Yes I want to mount a local directory into the Docker container so that I can run commands against my project source code. 

I'll give this a try and update the docs to include instructions if it work out.
